src/rpcblockchain.cpp

```diff
@@ -392,11 +394,36 @@ UniValue getblock(const UniValue& params, bool fHelp)
             "\nExamples:\n"
             + HelpExampleCli("getblock", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\"")
             + HelpExampleRpc("getblock", "\"00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09\"")
+            + HelpExampleCli("getblock", "12800")
+            + HelpExampleRpc("getblock", "12800")
         );
 
     LOCK(cs_main);
 
     std::string strHash = params[0].get_str();
+
+	// If height is supplied, find the hash
+	if (strHash.size() < 64) {
+	    // std::stoi allows characters, whereas we want to be strict
+	    regex r("[[:digit:]]+");
+	    if (!regex_match(strHash, r)) {
+	        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block height parameter");
+	    }
+        
+        int nHeight = -1;
+        try {
+            nHeight = std::stoi(strHash);
+	}
+	catch (const std::exception &e) {
+	    throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid block height parameter");
+	}
```

Comment:
I still see an indentation problem - lines 416-419 start with a tab instead of eight spaces.